tests/src/Kernel/Entity/EntityGroupsTest.php

```diff
@@ -0,0 +1,216 @@
+  }
+
+  /**
+   * Tests owner groups are returned correctly.
+   */
+  public function testOwnerGroupsOnly() {
+    $actual = Og::getEntityGroups('user', $this->user1->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+
+    // Also check isMember.
+    $this->assertTrue(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user1->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user2->id()));
+
+    $actual = Og::getEntityGroups('user', $this->user2->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group2, $actual);
+
+    // Also check isMember.
+    $this->assertTrue(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user2->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user1->id()));
+  }
+
+  /**
+   * Tests other groups users are added to.
+   */
+  public function testOtherGroups() {
+    // Should be a part of no groups.
+    $this->assertEquals([], Og::getEntityGroups('user', $this->user3->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user3->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user3->id()));
+
+    Og::invalidateCache();
+
+    // Add to group 1 should return that.
+    $this->createMembership($this->user3, $this->group1);
+
+    $actual = Og::getEntityGroups('user', $this->user3->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+
+    $this->assertTrue(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user3->id()));
+    $this->assertFalse(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user3->id()));
+
+    Og::invalidateCache();
+
+    // Add to group 2 should also return that.
+    $this->createMembership($this->user3, $this->group2);
+
+    $actual = Og::getEntityGroups('user', $this->user3->id());
+
+    $this->assertArrayHasKey('entity_test', $actual);
+    $this->assertCount(2, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+    $this->assertGroupExistsInResults($this->group2, $actual);
+
+    $this->assertTrue(Og::isMember('entity_test', $this->group1->id(), 'user', $this->user3->id()));
+    $this->assertTrue(Og::isMember('entity_test', $this->group2->id(), 'user', $this->user3->id()));
+  }
+
+  /**
+   * Creates an Og membership entity,
+   *
+   * @param \Drupal\user\Entity\User $user
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *
+   * @return \Drupal\og\Entity|OgMembership
+   */
+  protected function createMembership($user, $group) {
```

Comment:
Just as a note: This is a temp function, which will be later replaced by `Og::group()` (the port of `og_group`). In D7, calling `og_membership_create()` directly was discouraged, as it didn't do all the necessary access check, and association with the right group audience field.